### PR TITLE
Add data dicts for hubble api docs

### PR DESCRIPTION
**Pull request by chowbao**
Adding data dictionaries for hubble api docs

TODO: organize and add correct api doc formatting/structure to each mdx

### CODE REVIEW

File: api/hubble/account_signers.mdx [hunk 1]
```diff
@@ -0,0 +1,10 @@
...
+| deleted | bool | Indicates whether the ledger entry (account, claimable balance, trust line, offer, liquidity pool) has been deleted or not. Once an entry is deleted, it cannot be recovered.<br /><br />All state tables maintain history for deleted ledger entries.<br /><br />- Required Field |
+| batch_id | string | String representation of the run id for a given DAG in Airflow. Takes the form of "scheduled__<batch_end_date>-<dag_alias>". Batch ids are unique to the batch and help with monitoring and rerun capabilities |
+| batch_run_date | datetime | The start date for the batch interval. When taken with the date in the batch_id, the date represents the interval of ledgers processed. The batch run date can be seen as a proxy of closed_at for a ledger. |
+| batch_insert_ts | timestamp | The timestamp in UTC when a batch of records was inserted into the database. This field can help identify if a batch executed in real time or as part of a backfill. The timestamp should not be used during ad hoc analysis and is useful for data engineering purposes. |
```
Comment (sydneynotthecity) [inline api/hubble/account_signers.mdx:10]: What do you think about us dropping the batch fields from the docs? Since they really don't serve a purpose to the community, I wonder if we should just drop them from docs or make a one time note that `batch_id` and `batch_insert_ts` shouldn't be used

File: api/hubble/accounts.mdx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+| num_subentries | int64 | The total number of ledger entries connected to this account. Ledger entries include: trustlines, offers, signers, and data entries. (Claimable balances are counted under sponsoring entries, not subentries). Any newly created trustline, offer, signer or data entry will increase the number of subentries by 1. Accounts may have up to 1,000 subentries<br /><br />- Required Field<br /><br />#### Notes:<br />Each entry on a ledger takes up space, which is expensive to store on the blockchain. For each entry, an account is required to hold a [minimum XLM balance](https://developers.stellar.org/docs/fundamentals-and-concepts/lumens#minimum-balance). The reserve is calculated by (2 + num_subentries - num_sponsoring + num_sponsored) * 0.5XLM |
+| inflation_destination | string | Deprecated: The account address to receive an inflation payment when they are disbursed on the network.<br /><br />#### Notes:<br />Inflation was discontinued in 2019 by validator vote. |
```
Comment (sydneynotthecity) [inline api/hubble/accounts.mdx:9]: Should we bold **deprecated**?

File: api/hubble/accounts.mdx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+| home_domain | string | The domain that hosts this account's stellar.toml file.<br /><br />- Required Field<br /><br />#### Notes:<br />Only applies to asset issuer accounts. The stellar.toml file contains metadata about the asset issuer which helps identify who the issuer is and instills trust in the asset |
```
Comment (sydneynotthecity) [inline api/hubble/accounts.mdx:11]: nit: i don't think home domain is actually required. we might want to double check that.

File: api/hubble/accounts.mdx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+| batch_id | string | String representation of the run id for a given DAG in Airflow. Takes the form of "scheduled__<batch_end_date>-<dag_alias>". Batch ids are unique to the batch and help with monitoring and rerun capabilities |
+| batch_run_date | datetime | The start date for the batch interval. When taken with the date in the batch_id, the date represents the interval of ledgers processed. The batch run date can be seen as a proxy of closed_at for a ledger. |
+| batch_insert_ts | timestamp | The timestamp in UTC when a batch of records was inserted into the database. This field can help identify if a batch executed in real time or as part of a backfill. The timestamp should not be used during ad hoc analysis and is useful for data engineering purposes. |
+| sponsor | string | The account address of the sponsor who is paying the reserves for this ledger entry.<br /><br />The following ledger entry types can be sponsored:<br /><br /> - accounts<br /> - account signers<br /> - claimable balances<br /> - trust lines<br /><br />#### Notes:<br />Sponsors of claimable balances are the accounts that created the balance. |
```
Comment (sydneynotthecity) [inline api/hubble/accounts.mdx:22]: Even though these fields are appended on to the end of the table, does it make sense to move up past batch fields?

File: api/hubble/claimable_balances.mdx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+| claimants.predicate.abs_before | string | Deadline for when the balance must be claimed. If a balance is claimed before the date then the clause of the condition is satisfied. |
+| claimants.predicate.abs_before_epoch | int64 | A UNIX epoch value in seconds representing the same deadline date as abs_before. |
+| claimants.predicate.rel_before | int64 | A relative deadline for when the claimable balance can be claimed. The value represents the number of seconds since the close time of the ledger which created the claimable balance<br /><br />#### Notes:<br />This condition is useful when creating a timebounds based on creation conditions. If the creator wanted a balance only claimable one week after creation, this condition would satisfy that rule. |
+| claimants.predicate.unconditional | bool | If true it means this clause of the condition is always satisfied.<br /><br />#### Notes:<br />When the predicate is only unconditional = true, it means that the balance can be claimed under any conditions |
```
Comment (sydneynotthecity) [inline api/hubble/claimable_balances.mdx:10]: The notes here sound awkward. Would reword slightly

File: api/hubble/history_assets.mdx
Comment (sydneynotthecity) [inline api/hubble/history_assets.mdx:1]: are we missing `asset_id` for this table?

File: api/hubble/enriched_history_operations.mdx [hunk 6]
```diff
@@ -0,0 +1,134 @@
+| Column Name | Type | Description |
+| --- | --- | --- |
+| op_id | int64 | Unique identifier for an operation.<br /><br />- Primary Key<br />- Natural Key<br />- Cluster Field<br />- Required Field<br /><br />#### Notes:<br />The operation id is the transaction id + order number |
+| op_source_account | string | The account address that originates the operation<br /><br />- Cluster Field<br />- Required Field<br /><br />#### Notes:<br />Defaults to '' |
```
Comment (sydneynotthecity) [inline api/hubble/enriched_history_operations.mdx:4]: i don't think this field is ever ''

File: api/hubble/enriched_history_operations.mdx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+| op_source_account_muxed | string | If an account is multiplexed (muxed), the virtual account address that originates the operation |
+| transaction_id | int64 | A unique identifier for this transaction.<br /><br />- Primary Key<br />- Natural Key<br />- Cluster Field<br />- Required Field |
+| type | int64 | The number indicating which type of operation this operation executes<br /><br />- Required Field |
```
Comment (sydneynotthecity) [inline api/hubble/enriched_history_operations.mdx:7]: `type` is also a cluster field

File: api/hubble/enriched_history_operations.mdx [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+| type_string | string | The type of operation this operation executes<br /><br />- Required Field |
```
Comment (sydneynotthecity) [inline api/hubble/enriched_history_operations.mdx:8]: I think this would be better as "String representation of the operation executed"

File: api/hubble/enriched_history_operations.mdx [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+| account | string | The amount of sold asset that was moved from the seller account to the buyer account, reported in terms of the sold amount<br /><br />### Only exists for the following operations:<br /><br /><table> <thead> <tr> <th>Type</th> <th>Operation</th> </tr> </thead> <tbody> <tr> <td>0</td> <td>Create Account</td> </tr> <tr> <td>8</td> <td>Account Merge</td> </tr> </tbody> </table> |
```
Comment (sydneynotthecity) [inline api/hubble/enriched_history_operations.mdx:9]: copy/paste error?

File: api/hubble/history_trades.mdx [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+| rounding_slippage | int64 | Applies to liquidity pool trades only. With fractional amounts of an asset traded, the network must round a fraction to the nearest whole number. This can cause the trade to "slip" price by a percentage compared with the original offer. Rounding slippage reports the percentage that dust trades slip before executing.<br /><br />#### Notes:<br />Defaults to 1<br />Rounding Slippage is always unprofitable for the trader and is not a valid way to try and extract more value from the network. |
+| seller_is_exact | bool | Indicates whether the buying or selling party trade was impacted by rounding slippage. If true, the buyer was impacted. If false, the seller was impacted |
+| selling_asset_id | int64 | Unique identifier for the asset code, type and issuer combination. This is not a primary key on the table<br /><br />- Natural Key<br />- Required |
+| buying_asset_id | int64 | Unique identifier for the asset code, type and issuer combination. This is not a primary key on the table<br /><br />- Natural Key<br />- Required |
```
Comment (sydneynotthecity) [inline api/hubble/history_trades.mdx:28]: Might be worthwhile to point out that joining on asset_id is more efficient since it's an integer. That's the only reason why we added these fields onto the table

File: api/hubble/history_transactions.mdx [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+| time_bounds | string | A transaction precondition that can be set to determine when a transaction is valid. The user can set a lower and upper timebound, defined as a UNIX timestamp when the transaction can be executed. <br />If the transaction attempts to execute outside of the time range, the transaction will fail |
+| successful | bool | Indicates if this transaction was successful or not<br /><br />#### Notes:<br />A transaction's success does not indicate whether it was included and written to a ledger. It only indicates whether the operations in the transaction were successfully applied to mutate the ledger state. |
+| fee_charged | int64 | The fee (in stroops) paid by the source account to apply this transaction to the ledger. At minimum, a transaction is charged # of operations * base fee. The minimum base fee is 100 stroops<br /><br />#### Notes:<br />The stroop is the fractional representation of a lumen (XLM). 1 stroop is 0.0000001 XLM. |
+| inner_transaction_hash | string | AAAA |
```
Comment (sydneynotthecity) [inline api/hubble/history_transactions.mdx:16]: todo: needs definition

File: api/hubble/history_transactions.mdx [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+| min_account_sequence_age | int64 | A transaction precondition that can be set to determine valid conditions for a transaction to be submitted to the network. <br />This condition contains a minimum duration of time that must have passed since the source account's sequence number changed for the transaction to be valid |
+| min_account_sequence_ledger_gap | int64 | A transaction precondition that can be set to determine valid conditions for a transaction to be submitted to the network. <br />This condition contains an integer representation of the minimum number of ledgers that must have closed since the source account's sequence number change for the transaction to be valid |
+| extra_signers | record | An array of up to two additional signers that must have corresponding signatures for this transaction to be valid |
+| tx_envelope | string | AAAA |
+| tx_result | string | AAAA |
+| tx_meta | string | AAAA |
+| tx_fee_meta | string | AAAA |
```
Comment (sydneynotthecity) [inline api/hubble/history_transactions.mdx:29]: todo: add definitions

File: api/hubble/liquidity_pools.mdx
Comment (sydneynotthecity) [inline api/hubble/liquidity_pools.mdx:1]: One thing that might be good for us to note in the table description is if a liquidity pool is deleted and the _same_ asset pair is created in a new pool, the pool will have the same pool id as the original pool.

File: api/hubble/trust_lines.mdx [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+| balance | float64 | The number of units of an asset held by this account.<br /><br />- Required Field |
+| trust_line_limit | int64 | The maximum amount of this asset that this account is willing to accept. The limit is specified when opening a trust line.<br /><br />- Required Field |
+| buying_liabilities | float64 | The sum of all buy offers owned by this account for XLM only<br /><br />- Required Field<br /><br />#### Notes:<br />The `accounts` table only reports monetary balances for XLM. Any other asset class is reported in the `trust_lines` table. For buy offers, the account must hold the amount of asset to complete the transaction |
+| selling_liabilities | float64 | The sum of all sell offers owned by this account for XLM only<br /><br />- Required Field<br /><br />#### Notes:<br />The `accounts` table only reports monetary balances for XLM. Any other asset class is reported in the `trust_lines` table. |
```
Comment (sydneynotthecity) [inline api/hubble/trust_lines.mdx:12]: copy/paste errors from the `accounts` tables

**Review by sydneynotthecity — COMMENTED** [review]
i'll help with the formatting of the actual docs tomorrow, this is a great first pass and will help make formatting tomorrow easy :) 

i didn't review the larger tables (`history_operations` and `enriched_history_operations`) because those will be easier to review when i'm able to preview the docs instead of reading raw markdown

**Comment by chowbao** [comment]
Close in favor of newer PR https://github.com/stellar/stellar-docs/pull/670/files
